### DISCUSSION TITLE
Add support for user-defined labels

### DISCUSSION THREAD

**Discussion by Alkarex** [jocmp/capyreader#531] [Feature Requests]
Hello,
In the GReader API (implemented by FreshRSS), there is the concept of *user labels* (also called tags sometimes), which would be nice to support in CapyReader. It is not the same concept than *article tags*, which are defined by the original Web site.

A user can add labels to articles to classify them or [to re-share them](https://freshrss.github.io/FreshRSS/en/users/user_queries.html#share-your-user-queries).
Labels can also be added automatically by FreshRSS on incoming articles.

An article may appear in a feed as well as in multiple labels (and potentially also in the "read later" / favourite list).

Here is a screenshot from News+ as an illustration:
![Screenshot_20200229-134519_News+](https://user-images.githubusercontent.com/1008324/75607792-e9db2b00-5afa-11ea-9a29-205bb5ce0b0f.png)

Do not hesitate to ask if if you need any additional documentation / examples.

This is the main feature missing from the [list of apps compatible with FreshRSS](https://github.com/FreshRSS/FreshRSS/blob/edge/README.md#apis--native-apps) before we can [recommend an app](https://github.com/FreshRSS/FreshRSS/issues/6382#issuecomment-2077855339).

Alkarex, from [FreshRSS](https://github.com/FreshRSS/FreshRSS)

**Comment by jocmp** [comment]
Hey @Alkarex, I have a first pass of user labels available in a dev release [2025.01.1099-dev](https://github.com/jocmp/capyreader/releases/tag/2025.01.1099-dev). Please let me know what you think! I'm excited to finally have this as a feature of the app.

**Reply by Alkarex** [reply]
Hello @jocmp and thanks for the update!
I am delighted to see your progress. The quick tests I have done seem to work fine so far 👍🏻 
Of course, I am looking forward to also have the possibility to edit user-defined labels of a given article (add, remove) :-)

**Reply by jocmp** [reply]
Good to hear it's working! Thanks for trying it out.

I'll track the ability to add and remove labels here

- #786 
- #787

**Comment by regystro** [comment]
Hi. Thank you very much, I was looking for this feature and I really appreciate it.

Would it be possible to display a counter in Capy Reader showing unread items in the “My Labels”/ "Searches" / "User labels" left panel, as is done in the other categories/folders and the FreshRSS web interface?

<img width="285" height="317" alt="image" src="https://github.com/user-attachments/assets/02d4cf8f-30a2-41bb-84b8-224b53fc3380" />

This way, you can quickly check whether you have any unread filtered items that may be useful to you.

Thank you :)

**Reply by Alkarex** [reply]
For the record, this information is exposed in the GReader API as implemented by FreshRSS via the `unread_count` attribute associated to the label/tag
https://github.com/FreshRSS/FreshRSS/blob/cc32b2daa5d47a465eaa8d9b5d632610d473aedf/p/api/greader.php#L307

**Reply by jocmp** [reply]
Thanks @Alkarex! As it's implemented right now, I'm grabbing the unread count from the device-local SQLite database which is mostly correct (though could vary). @regystro it's a good idea, I don't remember why I didn't add this in initially.

**Reply by jocmp** [reply]
These counts are available in the latest version, [2026.02.1189](https://github.com/jocmp/capyreader/releases/tag/2026.02.1189).

<img width="300" alt="Labels" src="https://github.com/user-attachments/assets/54e02b64-f5bb-4122-8fc5-a25dd37c60fc" />
